### ISSUE TITLE
Crash on Linux for `run` Command with `--target` Because Docker Tool is not Verified Before Use

### ISSUE THREAD

**Issue by rmartin16** [beeware/briefcase#1207]
### Describe the bug

The `briefcase run linux system --target <target>` command crashes. Releases `0.3.13` and `0.3.14` are affected.

```
❯ briefcase -V
0.3.14

❯ briefcase run --target ubuntu

[helloworld] Finalizing application configuration...
Checking Docker target image ubuntu...

Log saved to /home/russell/tmp/beeware/helloworld/logs/briefcase.2023_04_23-18_38_02.run.log

Traceback (most recent call last):
  File "/home/russell/tmp/beeware/venv-del/bin/briefcase", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/russell/tmp/beeware/venv-del/lib/python3.11/site-packages/briefcase/__main__.py", line 25, in main
    command(**options)
  File "/home/russell/tmp/beeware/venv-del/lib/python3.11/site-packages/briefcase/commands/run.py", line 273, in __call__
    self.finalize(app)
  File "/home/russell/tmp/beeware/venv-del/lib/python3.11/site-packages/briefcase/commands/base.py", line 529, in finalize
    self.finalize_app_config(app)
  File "/home/russell/tmp/beeware/venv-del/lib/python3.11/site-packages/briefcase/platforms/linux/system.py", line 188, in finalize_app_config
    self.tools.docker.prepare(app.target_image)
    ^^^^^^^^^^^^^^^^^
AttributeError: 'ToolCache' object has no attribute 'docker'
```

### Steps to reproduce

`briefcase new --no-input && cd helloworld && briefcase run --target ubuntu`

### Expected behavior

Briefcase should verify the `Docker` tool before it is used in `LinuxSystemPassiveMixin.finalize_app_config()`.

### Environment

- Operating System: ubuntu 23.04
- Python version: 3.11
- Software versions:
  - Briefcase: `0.3.13`, `0.3.14`, `0.3.15.dev50+g2d19e158`

### Logs

<details>

<summary>Briefcase Log File</summary>

```
Date/Time:       2023-04-23 18:53:26 
Command line:    /home/user/github/beeware/briefcase/venv-3.11-briefcase/bin/briefcase run --target ubuntu

OS Release:      Linux 6.2.0-20-generic
OS Version:      #20-Ubuntu SMP PREEMPT_DYNAMIC Thu Apr  6 07:48:48 UTC 2023
Architecture:    x86_64
Platform:        Linux-6.2.0-20-generic-x86_64-with-glibc2.37

Python exe:      /home/user/github/beeware/briefcase/venv-3.11-briefcase/bin/python
Python version:  3.11.2 (main, Mar 13 2023, 12:18:29) [GCC 12.2.0]
Virtual env:     True
Conda env:       False

Briefcase:       0.3.14
Target platform: linux
Target format:   system

Environment Variables:
    COLORTERM=truecolor
    DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1000/bus
    DEBUGINFOD_URLS=https://debuginfod.ubuntu.com 
    DESKTOP_SESSION=ubuntu
    DISPLAY=:0
    GDMSESSION=ubuntu
    GNOME_DESKTOP_SESSION_ID=this-is-deprecated
    GNOME_SETUP_DISPLAY=:1
    GNOME_SHELL_SESSION_MODE=ubuntu
    GNOME_TERMINAL_SCREEN=/org/gnome/Terminal/screen/487c785f_788e_4a39_b40d_c5bd527a552c
    GNOME_TERMINAL_SERVICE=:1.132
    GTK_MODULES=gail:atk-bridge
    HOME=/home/user
    IM_CONFIG_PHASE=1
    LANG=en_US.UTF-8
    LESSCLOSE=/usr/bin/lesspipe %s %s
    LESSOPEN=| /usr/bin/lesspipe %s
    LOGNAME=user
    LS_COLORS=rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=00:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.avif=01;35:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.webp=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:*~=00;90:*#=00;90:*.bak=00;90:*.old=00;90:*.orig=00;90:*.part=00;90:*.rej=00;90:*.swp=00;90:*.tmp=00;90:*.dpkg-dist=00;90:*.dpkg-old=00;90:*.ucf-dist=00;90:*.ucf-new=00;90:*.ucf-old=00;90:*.rpmnew=00;90:*.rpmorig=00;90:*.rpmsave=00;90:
    OLDPWD=/home/user/github/beeware/briefcase/tmp
    PATH=/home/user/github/beeware/briefcase/venv-3.11-briefcase/bin:/home/user/github/beeware/briefcase/venv-3.11-briefcase/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin
    PS1=(venv-3.11-briefcase) \[\e]0;\u@\h: \w\a\]${debian_chroot:+($debian_chroot)}\[\033[01;32m\]\u@\h\[\033[00m\]:\[\033[01;34m\]\w\[\033[00m\]\$ 
    PWD=/home/user/github/beeware/briefcase/tmp/helloworld
    QT_ACCESSIBILITY=1
    QT_IM_MODULE=ibus
    SESSION_MANAGER=local/vm-ubuntu2304:@/tmp/.ICE-unix/1388,unix/vm-ubuntu2304:/tmp/.ICE-unix/1388
    SHELL=/bin/bash
    SHLVL=2
    SSH_AGENT_LAUNCHER=openssh
    SSH_AUTH_SOCK=/run/user/1000/keyring/ssh
    SYSTEMD_EXEC_PID=1409
    TERM=xterm-256color
    USER=user
    USERNAME=user
    VIRTUAL_ENV=/home/user/github/beeware/briefcase/venv-3.11-briefcase
    VIRTUAL_ENV_PROMPT=(venv-3.11-briefcase) 
    VTE_VERSION=7200
    WAYLAND_DISPLAY=wayland-0
    XAUTHORITY=/run/user/1000/.mutter-Xwaylandauth.O6WP31
    XDG_CONFIG_DIRS=/etc/xdg/xdg-ubuntu:/etc/xdg
    XDG_CURRENT_DESKTOP=ubuntu:GNOME
    XDG_DATA_DIRS=/usr/share/ubuntu:/usr/share/gnome:/usr/local/share/:/usr/share/:/var/lib/snapd/desktop
    XDG_MENU_PREFIX=gnome-
    XDG_RUNTIME_DIR=/run/user/1000
    XDG_SESSION_CLASS=user
    XDG_SESSION_DESKTOP=ubuntu
    XDG_SESSION_TYPE=wayland
    XMODIFIERS=@im=ibus
    _=/home/user/github/beeware/briefcase/venv-3.11-briefcase/bin/briefcase

Briefcase Log:
[18:53:26]                                                                                                                                                             system.py:181
           [helloworld] Finalizing application configuration...                                                                                                        system.py:181
           Checking Docker target image ubuntu...                                                                                                                      system.py:187

Main thread traceback:
╭─────────────────────────────────────────────────────────────────────── Traceback (most recent call last) ────────────────────────────────────────────────────────────────────────╮
│ /home/user/github/beeware/briefcase/src/briefcase/__main__.py:25 in main                                                                                                         │
│                                                                                                                                                                                  │
│   22 │   │   command = Command(logger=logger, console=console)                                                                                                                   │
│   23 │   │   options = command.parse_options(extra=extra_cmdline)                                                                                                                │
│   24 │   │   command.parse_config(Path.cwd() / "pyproject.toml")                                                                                                                 │
│ ❱ 25 │   │   command(**options)                                                                                                                                                  │
│   26 │   except HelpText as e:                                                                                                                                                   │
│   27 │   │   logger.info()                                                                                                                                                       │
│   28 │   │   logger.info(str(e))                                                                                                                                                 │
│                                                                                                                                                                                  │
│ ╭──────────────────────────────────────────────────────────────────────────────── locals ────────────────────────────────────────────────────────────────────────────────╮       │
│ │       command = <briefcase.platforms.linux.system.LinuxSystemRunCommand object at 0x7f4c0bd07950>                                                                      │       │
│ │       Command = <class 'briefcase.platforms.linux.system.LinuxSystemRunCommand'>                                                                                       │       │
│ │       console = <briefcase.console.Console object at 0x7f4c0c115690>                                                                                                   │       │
│ │ extra_cmdline = ['--target', 'ubuntu']                                                                                                                                 │       │
│ │        logger = <briefcase.console.Log object at 0x7f4c0c117690>                                                                                                       │       │
│ │       options = {'appname': None, 'update': False, 'update_requirements': False, 'update_resources': False, 'no_update': False, 'test_mode': False, 'passthrough': []} │       │
│ │        result = 0                                                                                                                                                      │       │
│ ╰────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯       │
│                                                                                                                                                                                  │
│ /home/user/github/beeware/briefcase/src/briefcase/commands/run.py:273 in __call__                                                                                                │
│                                                                                                                                                                                  │
│   270 │   │                                                                                                                                                                      │
│   271 │   │   # Confirm host compatibility, that all required tools are available,                                                                                               │
│   272 │   │   # and that the app configuration is finalized.                                                                                                                     │
│ ❱ 273 │   │   self.finalize(app)                                                                                                                                                 │
│   274 │   │                                                                                                                                                                      │
│   275 │   │   template_file = self.bundle_path(app)                                                                                                                              │
│   276 │   │   binary_file = self.binary_path(app)                                                                                                                                │
│                                                                                                                                                                                  │
│ ╭──────────────────────────────────────────────── locals ─────────────────────────────────────────────────╮                                                                      │
│ │                 app = <com.example.helloworld v0.0.1 AppConfig>                                         │                                                                      │
│ │             appname = None                                                                              │                                                                      │
│ │           no_update = False                                                                             │                                                                      │
│ │             options = {}                                                                                │                                                                      │
│ │         passthrough = []                                                                                │                                                                      │
│ │                self = <briefcase.platforms.linux.system.LinuxSystemRunCommand object at 0x7f4c0bd07950> │                                                                      │
│ │           test_mode = False                                                                             │                                                                      │
│ │              update = False                                                                             │                                                                      │
│ │ update_requirements = False                                                                             │                                                                      │
│ │    update_resources = False                                                                             │                                                                      │
│ ╰─────────────────────────────────────────────────────────────────────────────────────────────────────────╯                                                                      │
│                                                                                                                                                                                  │
│ /home/user/github/beeware/briefcase/src/briefcase/commands/base.py:529 in finalize                                                                                               │
│                                                                                                                                                                                  │
│   526 │   │   │   │   │   delattr(app, "__draft__")                                                                                                                              │
│   527 │   │   else:                                                                                                                                                              │
│   528 │   │   │   if hasattr(app, "__draft__"):                                                                                                                                  │
│ ❱ 529 │   │   │   │   self.finalize_app_config(app)                                                                                                                              │
│   530 │   │   │   │   delattr(app, "__draft__")                                                                                                                                  │
│   531 │                                                                                                                                                                          │
│   532 │   def verify_app_tools(self, app: BaseConfig):                                                                                                                           │
│                                                                                                                                                                                  │
│ ╭───────────────────────────────────────── locals ─────────────────────────────────────────╮                                                                                     │
│ │  app = <com.example.helloworld v0.0.1 AppConfig>                                         │                                                                                     │
│ │ self = <briefcase.platforms.linux.system.LinuxSystemRunCommand object at 0x7f4c0bd07950> │                                                                                     │
│ ╰──────────────────────────────────────────────────────────────────────────────────────────╯                                                                                     │
│                                                                                                                                                                                  │
│ /home/user/github/beeware/briefcase/src/briefcase/platforms/linux/system.py:188 in finalize_app_config                                                                           │
│                                                                                                                                                                                  │
│    185 │   │   │                                                                                                                                                                 │
│    186 │   │   │   # Ensure that the Docker base image is available.                                                                                                             │
│    187 │   │   │   self.logger.info(f"Checking Docker target image {app.target_image}...")                                                                                       │
│ ❱  188 │   │   │   self.tools.docker.prepare(app.target_image)                                                                                                                   │
│    189 │   │   │                                                                                                                                                                 │
│    190 │   │   │   # Extract release information from the image.                                                                                                                 │
│    191 │   │   │   output = self.tools.docker.check_output(                                                                                                                      │
│                                                                                                                                                                                  │
│ ╭───────────────────────────────────────── locals ─────────────────────────────────────────╮                                                                                     │
│ │  app = <com.example.helloworld v0.0.1 AppConfig>                                         │                                                                                     │
│ │ self = <briefcase.platforms.linux.system.LinuxSystemRunCommand object at 0x7f4c0bd07950> │                                                                                     │
│ ╰──────────────────────────────────────────────────────────────────────────────────────────╯                                                                                     │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
AttributeError: 'ToolCache' object has no attribute 'docker'
```
</details>

**Comment by rmartin16** [comment]
Using `briefcase run linux system` with `--target` feels like something that shouldn't be supported...unless we're going to do some magic with getting docker to talk to the local x server....but it should exit a bit more elegantly probably :)

**Comment by freakboy3742** [comment]
Agreed that `run --target` is an option that doesn't make much sense. We *might* be able to get it to work, but it's going to be complicated; so it's probably better to avoid it entirely.

**Comment by cguardia** [comment]
One way to fix this is to remove the ``--target`` option from the ``LinuxSystemPassiveMixin``, which right now is only used by the run command, and add it to the ``LinuxSystemMostlyPassiveMixin``.

**Comment by freakboy3742** [comment]
@cguardia I'd need to do a close check to be 100% sure, but this sounds plausible.

**Comment by cguardia** [comment]
Fixed by #1222.
